pydantic_ai_slim/pydantic_ai/models/mistral.py

```diff
@@ -509,6 +510,101 @@ def timestamp(self) -> datetime:
         return self._timestamp
 
 
+class _JSONChunkParser:
```

Comment:
Have you tried using `pydantic_core.from_json(<value>, allow_partial=True)`? If that doesn't handle cases that arise in practice with this model, it would be useful to see some examples it doesn't handle the way you want, as they might be things that we can improve in that function.

If it does accomplish what you need, then maybe we can remove this whole class